controller/authorize_blackbox_test.go

```diff
@@ -0,0 +1,301 @@
+	authorizeCtx, err = app.NewAuthorizeAuthorizeContext(goaCtx, req, goa.New("LoginService"))
+	require.NotNil(t, err)
+	require.Contains(t, err.Error(), "400 invalid_request: missing required parameter")
+	jsonapi.JSONErrorResponse(authorizeCtx, err)
+	require.Equal(t, 400, rw.Code)
+	prms.Add("redirect_uri", redirectURI)
+
+	prms.Del("client_id")
+	goaCtx = goa.NewContext(goa.WithAction(ctx, "AuthorizeTest"), rw, req, prms)
+	authorizeCtx, err = app.NewAuthorizeAuthorizeContext(goaCtx, req, goa.New("LoginService"))
+	require.NotNil(t, err)
+	require.Contains(t, err.Error(), "400 invalid_request: missing required parameter")
+	jsonapi.JSONErrorResponse(authorizeCtx, err)
+	require.Equal(t, 400, rw.Code)
+	prms.Add("client_id", rest.Configuration.GetPublicOauthClientID())
+
+	prms.Del("state")
+	goaCtx = goa.NewContext(goa.WithAction(ctx, "AuthorizeTest"), rw, req, prms)
+	authorizeCtx, err = app.NewAuthorizeAuthorizeContext(goaCtx, req, goa.New("LoginService"))
+	require.NotNil(t, err)
+	require.Contains(t, err.Error(), "400 invalid_request: missing required parameter")
+	jsonapi.JSONErrorResponse(authorizeCtx, err)
+	require.Equal(t, 400, rw.Code)
+}
+
+func (rest *TestAuthorizeREST) TestAuthorizeUnauthorizedError() {
+	t := rest.T()
+	svc, ctrl := rest.UnSecuredController()
+
+	redirect := "https://openshift.io"
+	clientID := ""
+	responseType := "code"
+	state := uuid.NewV4()
+
+	test.AuthorizeAuthorizeUnauthorized(t, svc.Context, svc, ctrl, nil, clientID, redirect, responseType, nil, state)
+}
+
+func (rest *TestAuthorizeREST) TestAuthorizeCallbackOK() {
+	t := rest.T()
+	_, ctrl := rest.UnSecuredController()
+
+	rw := httptest.NewRecorder()
+	u := &url.URL{
+		Path: fmt.Sprintf(client.AuthorizeAuthorizePath()),
+	}
+	req, err := http.NewRequest("GET", u.String(), nil)
+	require.Nil(t, err)
+
+	redirectURI := "https://openshift.io/somepath"
+	prms := url.Values{}
+
+	prms.Add("response_type", "code")
+	prms.Add("redirect_uri", redirectURI)
+	prms.Add("client_id", rest.Configuration.GetPublicOauthClientID())
+	prms.Add("state", uuid.NewV4().String())
+
+	ctx := context.Background()
+	goaCtx := goa.NewContext(goa.WithAction(ctx, "AuthorizeTest"), rw, req, prms)
+	authorizeCtx, err := app.NewAuthorizeAuthorizeContext(goaCtx, req, goa.New("LoginService"))
+	require.Nil(t, err)
+
+	err = ctrl.Authorize(authorizeCtx)
+	require.Nil(t, err)
+
+	require.Equal(t, 307, rw.Code) // redirect to keycloak login page.
+
+	locationString := rw.HeaderMap["Location"][0]
+	locationUrl, err := url.Parse(locationString)
+	require.Nil(t, err)
+
+	allQueryParameters := locationUrl.Query()
+
+	require.NotNil(t, allQueryParameters)
+	require.NotNil(t, allQueryParameters["state"][0])
+
+	returnedState := allQueryParameters["state"][0]
+
+	u = &url.URL{
+		Path: fmt.Sprintf(client.CallbackAuthorizePath()),
+	}
+
+	prms = url.Values{
+		"state": {returnedState},
+		"code":  {"SOME_OAUTH2.0_CODE"},
```

Comment:
Generate a new code here instead of using a hardcoded value. And check it below.